manager/src/main/java/com/redhat/service/smartevents/manager/api/models/requests/ProcessorRequest.java

```diff
@@ -106,4 +107,17 @@ public Source getSource() {
     public void setSource(Source source) {
         this.source = source;
     }
+
+    public Gateway getGateway() {
+        if (action == null && source == null) {
+            throw new IllegalStateException("The action and the source are null in the ProcessorRequest.");
+        }
+        if (action != null && source != null) {
+            throw new IllegalStateException("The ProcessorRequest can have only the action or the source.");
+        }
```

Comment:
In theory this condition is already validated by `@ValidProcessorGateway` validator. I don't know if it makes sense to add this additional check here

Reply:
Yeah, the problem is that with this public method maybe somebody starts using it somewhere else. So it's a bit safe programming in this case. But I can remove the checks, no strong opinion from my side

Reply:
I would prefer to remove it, in general this condition is a sort of "sanity check" for this class so it should be added also for a `getSourece` / `getAction` or in other methods. Can you please verify if it is possible to make the public constructor with no args `protected`? It might be because of serialization but Jackson should be smart enough

Reply:
we use `new ProcessorRequest()` in some tests of other packages. I've removed the checks